docs/appnotes/0019-using-flow-profiles.md

```diff
@@ -0,0 +1,90 @@
+However if the content for the new edit is derived from multiple sources then there is a need to match the flows from the different sources based on the technical characteristics.
+Currently the TAMS API does not provide an easy method to do this, so it is necessary to compare all the flows in code to find the matches.
+The more sources involved in this process, the harder this becomes.
+
+## TAMS Profiles within the API
+
+The TAMS profile model is split into a number of stages:
+
+* An end-point to list and describe the available profiles supported by the store
+* The ability to supply a profile when creating a flow and the de-normalisation of the technical details at the point of creation
+* The ability to query flows for a given profile to easily and quickly find the required media type
+
+### Profiles endpoint
+
+It is possible to create and view profiles via the dedicated API calls under the TAMS service end point (/services/profiles).
+These profiles hold all the technical parameters required to create a flow in a single location.
+A system looking to ingest standardised content into TAMS would create content matching one or more profiles as defined in the endpoint.
+
+A profile should be treated as immutable, so once created it cannot be updated.
+Updating a profile would cause mismatches with flows which have been already created using that profile and so breaks the model for profiles.
+To update a profile a new one with a new ID should be created.
+
+The profiles endpoint has been designed to provide extensibility of the metadata through the same tags model as for flows and sources.
+This would enable implementations to store additional metadata such as encoding parameters alongside the TAMS profile metadata
+
+### Creating a flow using a profile
+
+When creating a flow the generating system has two options:
+
+1. Specify all the technical characteristics of the flow including the wrapper, codec and essence parameters alongside the non-technical parameters such as label and description
+2. Provide just the technical profile for the flow and non-technical parameters required
+
+When using the second option on submission of the create flow request, the store will then read the metadata for a given profile and use this to populate the metadata for the flow.
+This de-normalisation of the technical parameters means that on the read side the flows created via both mechanisms have the same technical parameters available.
+Additionally if the decision is made to change a standard profile in the future then this does not affect the existing flows as the parameters have already been replicated and should remain unchanged to reflect the media actually being stored.
+
+Flows that have been created from a profile will include the parameter indicating which profile they were created from.
+This differentiates them from the flows that have been created with the technical characteristics directly.
```

Comment:
Do we need to differentiate between flows that were created using a Profile and a Flow created using exactly the same Metadata as an existing Profile?
If we didn't then perhaps the store should detect this at creation time and set the Profile Id to match even if not supplied?
OR it could reject the request saying that this Flow "metadata" matches an existing Profile and therefore the profile should be used?
Just thinking out loud here.